src/main/java/school/hei/haapi/service/FeeService.java

```diff
@@ -1,17 +1,57 @@
 package school.hei.haapi.service;
 
+import java.util.List;
 import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Service;
 import school.hei.haapi.model.Fee;
+import school.hei.haapi.model.exception.BadRequestException;
+import school.hei.haapi.model.validator.FeeValidator;
 import school.hei.haapi.repository.FeeRepository;
 
 @Service
 @AllArgsConstructor
 public class FeeService {
 
   private final FeeRepository feeRepository;
+  private final FeeValidator feeValidator;
 
-  public Fee getByUserIdAndId(String userId, String feeId) {
-    return feeRepository.getByUserIdAndId(userId, feeId);
+  public Fee getByStudentIdAndFeeId(String userId, String feeId) {
+    return feeRepository.getByStudentIdAndId(userId, feeId);
+  }
+
+  public List<Fee> saveAll(String studentId, List<Fee> fees) {
```

Comment:
need to check that `fee.list() < EVENTBRIDGE_MAX_PUT_EVENTS`

Reply:
Why ? Are fees asynchronous ?